src/Controls/tests/UITests/Tests/Issues/_IssuesUITest.cs

```diff
@@ -9,15 +9,40 @@ public abstract class _IssuesUITest : UITestBase
 
 		protected override void FixtureSetup()
 		{
-			base.FixtureSetup();
-			NavigateToIssue(Issue);
+			int retries = 0;
+			while (true)
```

Comment:
This retry logic seems orthogonal to the issue of not logging enough information in case of an error. If we need to retry tests, does that not signal we have a flaw in the product or a flaw in the tests?

Retrying tests and pretending nothing bad happened strikes me as very risky. I'd be OK with a strategy that retries tests just to keep things going, but still causes the build to fail (because clearly something is wrong).

Reply:
(Having said that, I would not have any retry logic at all unless we think it's necessary, and it isn't clear to me that's the case.)

Reply:
This is not retry logic in the test case itself, this is in the setup of the test.  Since we are trying to be time conscious we are not always shutting down the app and restarting it in between test cases/classes. Because of this you could run into weird issues/cases when trying to run the next set of tests. This particular 'retry' will essentially only restart the app and start fresh.

UITests in general have always been relatively fragile and we are trying to avoid noise/flakiness (There are lots of places we have retries in the CI already btw...) which is why we wanted to add this retry. @PureWeen @rmarinho

Reply:
> This is not retry logic in the test case itself, this is in the setup of the test.

Ah that's good to know, I didn't realize that.

> Because of this you could run into weird issues/cases when trying to run the next set of tests. 

So this is intended to fix the case where the test suite fails because a test left things in a 'bad' state, thus accidentally causing the next test to fail? I think that's reasonable.

> There are lots of places we have retries in the CI already

Indeed, and it's OK as long as it's retrying stuff we truly don't care about in our tests, such as anything purely external to MAUI, such as downloading and installing dotnet.exe, starting emulators, etc.


Would it be possible to log the restart as some kind of warning that we could monitor externally? Ideally it of course shouldn't ever need to retry because that means our tests are doing naughty things.

Reply:
> So this is intended to fix the case where the test suite fails because a test left things in a 'bad' state, thus accidentally causing the next test to fail?
Yes that would be a prime example of what this would be addressing.

> Would it be possible to log the restart as some kind of warning that we could monitor externally?
I think that sounds reasonable, it looks like NUnit has Assert.Warn, I will try adding that to see what results that produces.